ios/zxing/iphone/ZXingWidget/Classes/OverlayView.m

```diff
@@ -42,44 +42,73 @@ - (id) initWithFrame:(CGRect)theFrame
     self = [super initWithFrame:theFrame];
     if( self ) {
         rectangleEnabled = isRectangleEnabled;
-        CGFloat rectSize = self.frame.size.width - kPadding * 2;
-        if (!oneDMode) {
-            cropRect = CGRectMake(kPadding, (self.frame.size.height - rectSize) / 2, rectSize, rectSize);
-        } else {
-            CGFloat rectSize2 = self.frame.size.height - kPadding * 2;
-            cropRect = CGRectMake(kPadding, kPadding, rectSize, rectSize2);		
-        }
-        
         self.backgroundColor = [UIColor clearColor];
         self.oneDMode = isOneDModeEnabled;
-        if (isCancelEnabled) {
+        
+        if (isCancelEnabled)
+        {
             UIButton *butt = [UIButton buttonWithType:UIButtonTypeRoundedRect]; 
             self.cancelButton = butt;
             [cancelButton setTitle:@"Cancel" forState:UIControlStateNormal];
-            if (oneDMode) {
-                [cancelButton setTransform:CGAffineTransformMakeRotation(M_PI/2)];
-                
-                [cancelButton setFrame:CGRectMake(20, 175, 45, 130)];
-            }
-            else {
-                CGSize theSize = CGSizeMake(100, 50);
-                CGRect theRect = CGRectMake((theFrame.size.width - theSize.width) / 2, cropRect.origin.y + cropRect.size.height + 20, theSize.width, theSize.height);
-                [cancelButton setFrame:theRect];
-                
-            }
-            
             [cancelButton addTarget:self action:@selector(cancel:) forControlEvents:UIControlEventTouchUpInside];
             [self addSubview:cancelButton];
             [self addSubview:imageView];
         }
         
-        if (overlay != nil) {
+        [self updateViewsWithFrame:theFrame];
+        
+        if (overlay != nil)
+        {
             [self addSubview:overlay];
         }
     }
     return self;
 }
 
+-(void)updateViewsWithFrame:(CGRect)newFrame
```

Comment:
Yes, we should definitely use https://github.com/TheLevelUp/ZXingObjC. Let me FT this and we go with the current one for now, but should still think about moving to the most recent one in future releases, although no prio right now. Thanks!